js/tji-formmodal-whoami-signup-donate.js

```diff
@@ -0,0 +1,342 @@
+  var panel = panel || this.panels[this.state.panel];
+  var $error = jQuery('<div class="tji-modal__form-panel-error" />').text(error_message);
+  this.ui.$modal.find('.js-formpanel-' + panel)
+    .find('fieldset')
+    .before($error);
+}
+
+TJISignupDonateFormModal.prototype.clear_fields = function() {
+  this.ui.$modal.find('form').trigger('reset')
+}
+
+TJISignupDonateFormModal.prototype.clear_messages = function() {
+  this.ui.$modal.find('.tji-modal__form-panel-error').remove();
+  this.ui.$modal.find('.tji-modal__form-panel-success').remove();
+}
+
+TJISignupDonateFormModal.prototype.set_data_and_validate = function() {
+  this.clear_messages();
+  
+  var data = this.ui.$modal.find('.js-formpanel-' + this.panels[this.state.panel]).find('[name]').serializeArray();
+  _.assign(this.state.data, _.mapValues(_.keyBy(data, 'name'), 'value'));
+
+  if(this.panels[this.state.panel] === 'whoami') {
+    this.state.data.whoami = (this.state.data.whoami === 'other') ? this.state.data.whoami_other : this.state.data.whoami;
+    if (!this.state.data.whoami) {
+      this.render_validation_error('Please let us know what your deal is?');
+      return false;
+    }
+  }
+  if(this.panels[this.state.panel] === 'newsletter') {
+    this.state.data.fname = this.state.data.fname.trim();
+    if(!this.state.data.fname) {
+      this.render_validation_error('Please enter your first name.');
+      return false;
+    }
+    if(!/\S+@\S+\.\S+/.test(this.state.data.email)) {
+      this.render_validation_error('Why did you enter a bunk email?');
+      return false;
+    }
+  }
+  if(this.panels[this.state.panel] === 'donate') {
+    this.state.data.donor_fname = this.state.data.donor_fname.trim();
+    if(!this.state.data.donor_fname) {
+      this.render_validation_error('Please enter the donor\'s first name.');
+      return false;
+    }
+    this.state.data.donor_lname = this.state.data.donor_lname.trim();
+    if(!this.state.data.donor_lname) {
+      this.render_validation_error('Please enter the donor\'s last name.');
+      return false;
+    }
+    if(!/\S+@\S+\.\S+/.test(this.state.data.donor_email)) {
+      this.render_validation_error('Why did you enter a bunk donor email?');
+      return false;
+    }
+    this.state.data.donation = this.state.data.donation === 'other' ? this.state.data.donation_other : this.state.data.donation;
+    this.state.data.donation = parseFloat(this.state.data.donation).toFixed(2);
+    if(isNaN(this.state.data.donation) || !this.state.data.donation) {
+      this.render_validation_error('Please enter a numeric dollar amount more than 0');
+      return false;
+    }
+  }
+
+  localStorage.setItem(this.local_storage_key, JSON.stringify(this.state.data));
+  return true;
+}
+
+TJISignupDonateFormModal.prototype.log = function() {
+  if(!this.set_data_and_validate())
+    return;
+
+  ga('send', 'event', 'whoami', this.state.data.whoami);
```

Comment:
Hey @LayaTaal yep -- 
open your developer tools
click on the 'application' tab
select 'local storage' from the left column menu 
select localhost / the host your on
select the key 'download'
click the 'x'

This will clear the localstorage item that is set that prevents the pop up from showing.

Reply:
Also I'm not sure how Google analytics is setup on our site -- on production is looks like ga is defined globally -- are there different setting for staging and localhost?

Reply:
That's a great tip for developer tools, thanks Simi!

For ga, let me look into this some more. There should not be different settings for staging and local.

Reply:
@LayaTaal Any updates on how GA is setup and how I can send ga events -- I was using this approach -- https://developers.google.com/analytics/devguides/collection/analyticsjs/events

Reply:
Hey @ifsimicoded, I looked into this more but I don't have a great answer. We are using Tag Manager, but since we are loading Analytics through Tag Manager, I think the method you are using should work fine. I solved the error I was getting (problem with my local setup). 

I'm not seeing the event come through our analytics though. My IP isn't filtered so I'm not sure why that might be. Will it only send events that come from the domain associated with the account? Any thoughts?

Other than this, I think the form looks really great!

Reply:
Hey thanks, I found some documentation about gtag and events -- give it a review again. I made some code updates to accommodate tag manager.  Alsoooo, how do you handle using your test Paypal account vs the production one -- in different environments?